baseapp/baseapp.go

```diff
@@ -133,6 +133,10 @@ type BaseApp struct { // nolint: maligned
 	// indexEvents defines the set of events in the form {eventType}.{attributeKey},
 	// which informs Tendermint what to index. If empty, all events will be indexed.
 	indexEvents map[string]struct{}
+
+	// msg fee handler
+	// custom fee handler for provenance, which will calculate and charge fee(if not run in simulation mode)
```

Comment:
Might want to reword this comment?  I assume that we are trying to upstream this change?  It seems like other chains may want the flexibility of a custom fee handler to be passed into the cosmos base app?

Reply:
for now i believe these are going to be provenance specific, and because of the middleware changes will not go into the sdk in their current form 🤷

Reply:
i'll take out the word provenance though ..good point

Reply:
👍 sounds good